sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -165,15 +187,19 @@ AZ_NODISCARD az_result az_iot_hub_client_sas_password_get(
  * @brief Gets the MQTT topic filter to subscribe to register responses.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
- *                              filter.
- * @param[out] out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @param[out] mqtt_topic_filter A buffer with sufficient capacity to hold the MQTT topic filter.
+ *                               If successful, contains a null-terminated string with the topic

```

Comment:
Similar docs feedback as https://github.com/Azure/azure-sdk-for-c/pull/568#discussion_r414145132

We should probably come up with some convention on documenting `az_result` returning APIs and apply it consistently.

I think providing the details on what `az_result` values can be returned and what they mean will be really helpful to callers. Whether we keep the similar description in the parameter docs, is up to you.

Reply:
That is non-trivial since we call other APIs and directly return their `az_result`.
@JeffreyRichter proposed that only results that should be handled by the application should be documented: e.g. if the incoming message parser attempted to parse something that was not meant for itself we can document the `AZ_ERROR_IOT_TOPIC_NO_MATCH`. 

Documenting common ones returned by all AZ_CORE won't scale and isn't really that useful since applications should treat them as unhandled exceptions (crash, report, restart) since usually there's nothing much that can be done to recover.